core/src/main/scala/quasar/destination/avalanche/AvalancheConfig.scala

```diff
@@ -79,20 +80,28 @@ object AvalancheConfig {
       ad => (ad.clientId.value, ad.tenantId.value, ad.clientSecret.value).some)(
       "clientId", "tenantId", "clientSecret")
 
-  implicit val avalancheConfigCodecJson: CodecJson[AvalancheConfig] =
-    casecodec5[String, String, URI, String, AzureCredentials.ActiveDirectory, AvalancheConfig](
-      (accountName, containerName, uri, password, creds) =>
-        AvalancheConfig(
-          AccountName(accountName),
-          ContainerName(containerName),
-          uri,
-          ClusterPassword(password),
-          creds),
-      cfg =>
-        (cfg.accountName.value,
-          cfg.containerName.value,
-          cfg.connectionUri,
-          cfg.password.value,
-          cfg.azureCredentials).some)(
-      "accountName", "containerName", "connectionUri", "clusterPassword", "credentials")
+  implicit def AvalancheConfigCodecJson: CodecJson[AvalancheConfig] =
+    CodecJson({ (c: AvalancheConfig) =>
+        ("accountName" := c.accountName.value) ->:
+        ("containerName" := c.containerName.value) ->:
+        ("connectionUri" := c.connectionUri) ->:
+        ("clusterPassword" := c.password.value) ->:
+        ((if (c.writeMode == Replace) { None } else { Some(c.writeMode) }).map("writeMode" := _)) ->?:
```

Comment:
Could we just always render the `writeMode`? Unless a consumer of the JSON is treating unexpected properties as an error, it shouldn't affect anything expecting the previous format. Do we know if we have any consumers like this?

Reply:
I prefer a clean roundtrip and prefer to have as few ways of encoding the same thing as possible. I don't feel strongly about this though so I can make it write it if thats preferable.

Reply:
Emrys pointed out that this roundtrip isn't clean in all cases anyways. Will adjust as he suggests.